src/api.ts

```diff
@@ -70,22 +70,28 @@ function getVisualizationFromLocalStorage(): PopulatedVisualization {
     const nonFilteredNodes = parentWindowOgma.getNonFilteredNodes();
     const nonFilteredEdges = parentWindowOgma.getNonFilteredEdges();
 
+    // getAttributes call is expensive, do the call only once for nodes and edges
+    const nonFilteredNodesAttributes = nonFilteredNodes.getAttributes();
+    const nonFilteredEdgesAttributes = nonFilteredEdges.getAttributes();
+
     const nodes = nonFilteredNodes.toJSON().map((n, index) => ({
       ...n,
-      attributes: nonFilteredNodes.getAttributes()[index]
+      attributes: nonFilteredNodesAttributes[index]
     })) as VizNode[];
 
     const edges = nonFilteredEdges.toJSON().map((e, index) => ({
       ...e,
-      attributes: nonFilteredEdges.getAttributes()[index]
+      attributes: nonFilteredEdgesAttributes[index]
     })) as VizEdge[];
 
     const visualization = JSON.parse(storeVisualizationData!) as Visualization;
-    return {
+
+    // We need to clone the object because attributes get mutated by the parent ogma
+    return structuredClone({
```

Comment:
since the `visualization` var is cloned at line 87, and that the each node/edge data is cloned implicitly when we call `.toJSON()` (on unfiltered nodes & edges), I think we could save a bit of resources by only cloning the attributes array at lines 74 & 75.

Not blocking, this works too.

Reply:
i am unsure about the performance impact of that; i prefer to move forward to unblock the release asap

We don't know how the contract might evolve in the future